src/job/MomoJob.ts

```diff
@@ -1,10 +1,53 @@
 export type Handler = () => Promise<string | undefined | void> | string | undefined | void;
 
-export interface MomoJob {
+interface MomoBaseJob {
   handler: Handler;
   name: string;
-  interval: string;
-  firstRunAfter?: number;
   concurrency?: number;
   maxRunning?: number;
 }
+
+export interface IntervalSchedule {
+  interval: string;
+  firstRunAfter: number;
+}
+
+export interface CronSchedule {
+  cronSchedule: string;
+}
+
+export function isIntervalSchedule(input: unknown): input is IntervalSchedule {
+  return (
+    // @ts-ignore
+    input.interval !== undefined &&
+    // @ts-ignore
+    input.firstRunAfter !== undefined &&
+    typeof (input as IntervalSchedule).interval == 'string' &&
+    typeof (input as IntervalSchedule).firstRunAfter == 'number'
+  );
+}
+
+export function isCronSchedule(input: unknown): input is CronSchedule {
+  return (
+    // @ts-ignore
+    input.cronSchedule !== undefined && typeof (input as CronSchedule).cronSchedule == 'string'
+  );
+}
+
+export interface MomoIntervalJob extends MomoBaseJob {
+  schedule: IntervalSchedule;
+}
+
+export function isMomoIntervalJob(input: unknown): input is MomoIntervalJob {
+  return isIntervalSchedule((input as MomoIntervalJob).schedule);
+}
+
+export interface MomoCronJob extends MomoBaseJob {
+  schedule: CronSchedule;
+}
+
+export function isMomoCronJob(input: MomoJob): input is MomoCronJob {
```

Comment:
this function is unused

removing the unused function should also allow you to simplify the code quite a bit, remove the MomoBaseJob, simply writing 
`export interface MomoJob {
  handler: Handler;
  name: string;
  schedule: Interval | CronSchedule;
  concurrency?: number;
  maxRunning?: number;
}`